source/common/http/http1/conn_pool.cc

```diff
@@ -300,11 +300,17 @@ ConnPoolImpl::PendingRequest::PendingRequest(ConnPoolImpl& parent, StreamDecoder
   parent_.host_->cluster().stats().upstream_rq_pending_total_.inc();
   parent_.host_->cluster().stats().upstream_rq_pending_active_.inc();
   parent_.host_->cluster().resourceManager(parent_.priority_).pendingRequests().inc();
```

Comment:
Currently, the circuit breaker state is updated on every `inc()` or `dec()` of the `Resource` itself, and is not explicitly related to the `overflow_` gauge. Do you still think we need to tie those together?

Reply:
oops sorry, I dont know why overflow gauge came out when I meant the resource gauge (not as in a stat, as in the int that tracks the concurrency in the Resource, as you pointed out). And yes, I still think that we should somehow tie the update of the resource and the update of the open gauge; because everytime we update the resource gauge we have the chance to be closing or opening the circuit breaker and thus the open gauge should be updated. However, if you feel this is too heavy handed, feel free to leave as is

Reply:
How about a callback that updates the circuit breaker state every time there is a `inc()` or `dec()` call?

Reply:
Yes please, in general, let's try to find a way to factor this so it's built into the resource manager and every callsite does not need updating. I'm not a fan of having to reimplement this logic every place that we need to do this.